core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -799,6 +834,13 @@ trait EntityComponent {
         EntityRecordRawSqlQuery.batchHide(workspaceContext.workspaceIdAsUUID, entRefs).map(res => res.sum)
     }
 
+    // "deletes" entities of a certain type by hiding and renaming them
+    def hideType(workspaceContext: Workspace, entityType: String): ReadWriteAction[Int] = {
+      EntityAndAttributesRawSqlQuery.batchHideAttributesOfType(workspaceContext, entityType) andThen
+        EntityRecordRawSqlQuery.batchHideType(workspaceContext.workspaceIdAsUUID, entityType).map(res => res.sum) andThen
+        workspaceQuery.updateLastModified(workspaceContext.workspaceIdAsUUID)
+    }
```

Comment:
```suggestion
    def hideType(workspaceContext: Workspace, entityType: String): ReadWriteAction[Int] = {
      for {
        _ <- EntityAndAttributesRawSqlQuery.batchHideAttributesOfType(workspaceContext, entityType)
        numEntitiesHidden <- EntityRecordRawSqlQuery.batchHideType(workspaceContext.workspaceIdAsUUID, entityType).map(res => res.sum)
        - <- workspaceQuery.updateLastModified(workspaceContext.workspaceIdAsUUID)
      } yield numEntitiesHidden
    }
```
as written, this method will now always return "1" on success - the number of rows changed by `workspaceQuery.updateLastModified` … I think you need to switch to the for/yield and control which value is returned

Reply:
gahhh nice catch! i'll handle this momentarily

Reply:
Fixed in 81f402ae4ad34408cf08a18cb5448f8a6fbf0cec . I also commit the unit test that I used to verify that it returned what I expected